cpp/src/gandiva/expr_decomposer.cc

```diff
@@ -212,11 +219,16 @@ Status ExprDecomposer::Visit(const LiteralNode& node) {
 //    that has a match will do it).
 // Both of the above optimisations save CPU cycles during expression evaluation.
 
-int ExprDecomposer::PushThenEntry(const IfNode& node) {
+int ExprDecomposer::PushThenEntry(const IfNode& node, bool reuse_bitmap) {
   int local_bitmap_idx;
 
-  if (!if_entries_stack_.empty() &&
-      if_entries_stack_.top()->entry_type_ == kStackEntryElse) {
+  if (reuse_bitmap) {
```

Comment:
the comment above this seems off - i am fine if we want to address this separare